api/wikis/permissions.py

```diff
@@ -39,7 +42,23 @@ def has_object_permission(self, request, view, obj):
 class ExcludeWithdrawalsWikiVersion(permissions.BasePermission):
 
     def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, WikiVersion), 'obj must be a WikiVersion, got {}'.format(obj)
         node = obj.wiki_page.node
         if node and node.is_retracted:
             return False
         return True
+
+
+class IsEnabled(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, (WikiPage, WikiVersion, AbstractNode)), 'obj must be a WikiPage, WikiVersion, or AbstractNode, got {}'.format(obj)
+        if isinstance(obj, WikiPage):
+            node = obj.node
+        elif isinstance(obj, WikiVersion):
+            node = obj.wiki_page.node
```

Comment:
For the WikiVersion endpoint(s), are we pre-fetching (i.e. select_related) `wiki_page` and `wiki_page.node`?

Reply:
Discussed in person: no, we aren't, but this check will only happen once for all of the versions endpoints (not n times).